mews-operations/availabilityBlockProcessor.md

```diff
@@ -0,0 +1,176 @@
+
+### Request
+
+[PlatformAddress]/api/channelManager/v1/processAvailabilityBlocks
+
+```json
+{
+    "clientToken": "[Channel manager client token]", 
+    "connectionToken": "[Token of a concrete connection]",
+    "messageId": "[Id of message which request relates to]",    
+    "availabilityBlock": {
+        "code": "33",
+        "confirmationNumber": "2",
+        "status": "Created",
+        "name": "SampleBlock",
+        "dates": {
+            "start": "2024-12-05",
+            "end": "2024-12-11"
+        },
+        "releaseStrategy": {
+            "type": "fixed",
+            "fixedDate": "2024-12-08"    
+        },
+        "originalRateCode": "FF",
+        "spaceCategories": [
+            {
+                "spaceTypeCode": "D",
+                "allocatedSpaces": [
+                    {
+                        "start": "2024-12-05",
+                        "end": "2024-12-07",
+                        "count": 2
+                    },
+                    {
+                        "start": "2024-12-08",
+                        "end": "2024-12-11",
+                        "count": 5
+                    }
+                ],
+                "rates": [
+                    {
+                        "start": "2024-12-05",
+                        "end": "2024-12-07",
+                        "grossAmount": 100.00
+                    },
+                    {
+                        "start": "2024-12-08",
+                        "end": "2024-12-11",
+                        "grossAmount": 50.00
+                    }
+                ]
+            }
+        ],
+        "booker": {
+            "email": "angelina.jolie@10001-438-sample-portfolio-hotel-10001.com",
+            "firstName": "Angelina",
+            "lastName": "Jolie",
+            "title": null,
+            "telephone": null,
+            "nationalityCode": null,
+            "languageCode": null,
+            "loyaltyCode": null,
+            "address": null
+        },
+        "company": null,
+        "notes": "This is a note."
+    }  
+}
+```
+
+| Property | Type | Contract | Description |
+| :-- | :-- | :-- | :-- |
+| `clientToken` | `string` | required | Client token of the channel manager. |
+| `connectionToken` | `string` | required | Token of a concrete connection. |
+| `messageId` | `string` | required | Unique identification of the message. Used for asynchronous confirmations. |
+| `availabilityBlock` | [`Availability block`](#availability-block) object | required | Availablity block data. |
+
+
+#### Availability block
+| Property | Type | Contract| Description |
+| :-- | :-- | :-- | :-- |
+| `code`| `string`| required | Unique reference code from external system for the availability block. Will be returned after confirmation containing this code. |
+| `confirmationNumber` | `string` | required(exc. create) | Mews confirmation number for the block. |
```

Comment:
I think it might be better to write this as follows:
```suggestion
| `confirmationNumber` | `string` | optional | Mews confirmation number for the block. If updating or deleting an existing block, this property is required. |
```